PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutScannerConfirmationFragment.java

```diff
@@ -9,4 +37,143 @@
+                (getActivity()).getFragmentManager().beginTransaction();
+        transaction.replace(R.id.checkout_activity_container, nextFrag, fragName);
+        transaction.addToBackStack(fragName);
+        transaction.commit();
+    }
+
+    /**
+     * Used to confirm the scan result
+     * Uses OnSubmit
+     */
+    // Why can't this be protected? wat
+    // this code is exactly like servicesscannerconfirmationfragmnent
+
+    class SearchByCodeResponseListener implements Response.Listener<JSONObject> {
+        @Override
+        public void onResponse(JSONObject jsonObject) {
+            try {
+                mRegistrationFound = jsonObject.getBoolean("status");
+                Log.d("found?", Boolean.toString(mRegistrationFound));
+            } catch (JSONException e) {
+                Log.e(TAG, "Error parsing JSON");
+                Log.e(TAG, e.toString());
+            }
+
+            if (mRegistrationFound) {
+                // Get services here
+                Log.d("found the user","");
+                try {
+                    serviceArray = jsonObject.getJSONArray("services");
+
+                    Bundle args = new Bundle();
+
+                    // Convert JSONArray to ArrayList
+                    ArrayList<String> mServices = new ArrayList<String>();
+                    if (serviceArray != null){
+                        for (int i=0; i < serviceArray.length(); i++){
+                            try {
+                                mServices.add(serviceArray.get(i).toString());
+                            } catch (JSONException e) {
+                                e.printStackTrace();
+                            }
+                        }
+                    }
+
+                    args.putStringArrayList("services",mServices);
+                    args.putCharSequence("scan_result", mScanResult);
+                    args.putBoolean("manual_input", mManualInput);
+
+                    CheckoutFormFragment formFrag = new CheckoutFormFragment();
+                    formFrag.setArguments(args);
+                    displayNextFragment(formFrag, CheckoutConfirmationFragment.TAG);
+                } catch (JSONException e) {
+                    Log.d(TAG,"error getting services");
+                    e.printStackTrace();
+                }
+
+            } else{
+                notFoundDialogue();
+            }
+
+        }
+    }
+
+    private class SearchByCodeErrorListener implements Response.ErrorListener{
+
+            @Override
+            public void onErrorResponse(VolleyError volleyError){
+                Log.e(TAG, "VolleyError.getLocalizedMessage = " + volleyError.getLocalizedMessage());
+                if (volleyError.getLocalizedMessage() != null){
+                    Log.e(TAG, "VolleyError.toString() " + volleyError.toString());
+                }
+                Toast toast = Toast.makeText(getActivity(), "Error looking up the QR Code",
```

Comment:
switch toast length to long, and change toast text to be more descriptive to the end user.
change the text to better distinguish it from the text in the notFoundDialogue - for example, something like "Server error looking up the QR code"